src/main/perl/RuleBasedEditor.pm

```diff
@@ -0,0 +1,1175 @@
+    value_fmt : value format (see LINE_VALUE_xxx constants)
+    value_opt : value interpretation/formatting options (bitmask, see LINE_VALUE_OPT_xxx constants)
+
+Return value:
+    A string corresponding to the value formatted according to the format specified by arguments
+    or undef in case of an internal error (missing arguments)
+
+=cut
+
+sub _formatAttributeValue
+{
+    my $function_name = "_formatAttributeValue";
+    my ($self, $attr_value, $line_fmt, $value_fmt, $value_opt) = @_;
+
+    unless (defined($attr_value)) {
+        *$self->{LOG}->error("$function_name: 'attr_value' argument missing (internal error)");
+        return undef;
+    }
+    unless (defined($line_fmt)) {
+        *$self->{LOG}->error("$function_name: 'list_fmt' argument missing (internal error)");
+        return undef;
+    }
+    unless (defined($value_fmt)) {
+        *$self->{LOG}->error("$function_name: 'value_fmt' argument missing (internal error)");
+        return undef;
+    }
+    unless (defined($value_opt)) {
+        *$self->{LOG}->error("$function_name: 'value_opt' argument missing (internal error)");
+        return undef;
+    }
+
+    *$self->{LOG}->debug(2,
+                         "$function_name: formatting attribute value >>>$attr_value<<< (line fmt=$line_fmt, value fmt=$value_fmt, value_opt=$value_opt)"
+                        );
+
+    my $formatted_value;
+    if ($value_fmt == LINE_VALUE_BOOLEAN) {
+        $formatted_value = $attr_value ? 'yes' : 'no';
+
+    } elsif ($value_fmt == LINE_VALUE_INSTANCE_PARAMS) {
+        $formatted_value = '';    # Don't return undef if no matching attributes is found
+                                  # Instance parameters are described in a nlist
+        $formatted_value .= " -l $attr_value->{logFile}"    if $attr_value->{logFile};
+        $formatted_value .= " -c $attr_value->{configFile}" if $attr_value->{configFile};
+        $formatted_value .= " -k $attr_value->{logKeep}"    if $attr_value->{logKeep};
+
+    } elsif ($value_fmt == LINE_VALUE_ARRAY) {
+        *$self->{LOG}->debug(2, "$function_name: array values received: ", join(",", @$attr_value));
+        if ($value_opt & LINE_VALUE_OPT_UNIQUE) {
+            my %values = map(($_ => 1), @$attr_value);
+            $attr_value = [keys(%values)];
+            *$self->{LOG}->debug(2, "$function_name: array values made unique: ", join(",", @$attr_value));
+        }
+        # LINE_VALUE_OPT_UNIQUE implies LINE_VALUE_OPT_SORTED
+        if ($value_opt & (LINE_VALUE_OPT_UNIQUE | LINE_VALUE_OPT_SORTED)) {
+            $attr_value = [sort(@$attr_value)] if $value_opt & (LINE_VALUE_OPT_UNIQUE | LINE_VALUE_OPT_SORTED);
+            *$self->{LOG}->debug(2, "$function_name: array values sorted: ", join(",", @$attr_value));
+        }
+        $formatted_value = join " ", @$attr_value;
+
+    } elsif ($value_fmt == LINE_VALUE_HASH_KEYS) {
+        $formatted_value = join " ", sort keys %$attr_value;
```

Comment:
This was by choice as it was matching the use case I met! We could always enrich this in the future with a value option to keep the value...